### PR TITLE
Implement FileIO.read with nbytes and importing modified imghdr.py

### PR DESCRIPTION
**Pull request by corona10**
- Update FileIO.read to support nbytes reading
- Import modified imghdr.py

### CODE REVIEW

File: Lib/imghdr.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+"""Recognize image file formats based on their first few bytes."""
+
+#from os import PathLike
```
Comment (coolreader18) [inline Lib/imghdr.py]: Generally our method for modifying stdlib modules is to add the unmodified module in one commit, and then modify what you need to in a separate commit, preferably marking the modification with `# XXX <description and reason for modification>`

File: vm/src/stdlib/io.rs [hunk 2]
```diff
@@ -346,19 +346,33 @@ fn file_io_init(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
...
 
     let mut bytes = vec![];
-    match handle.read_to_end(&mut bytes) {
-        Ok(_) => {}
-        Err(_) => return Err(vm.new_value_error("Error reading from Buffer".to_string())),
+    match read_byte {
+        None => {
+            match handle.read_to_end(&mut bytes) {
+                Ok(_) => {}
+                Err(_) => return Err(vm.new_value_error("Error reading from Buffer".to_string())),
```
Comment (coolreader18) [inline vm/src/stdlib/io.rs]: You can do:
```rust
handle.read_to_end(&mut bytes).map_err(|| vm.new_value_error(...))?
```

File: vm/src/stdlib/io.rs [hunk 3]
```diff
@@ -346,19 +346,33 @@ fn file_io_init(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
 }
 
 fn file_io_read(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(file_io, None)]);
+    arg_check!(vm, args, required = [(file_io, None)], optional = [(read_byte, None)]);
```
Comment (coolreader18) [inline vm/src/stdlib/io.rs]:
```suggestion
    arg_check!(vm, args, required = [(file_io, None)], optional = [(read_byte, Some(vm.ctx.int_type()))]);
```

File: vm/src/stdlib/io.rs [hunk 4]
```diff
@@ -346,19 +346,33 @@ fn file_io_init(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
...
+    match read_byte {
+        None => {
+            match handle.read_to_end(&mut bytes) {
+                Ok(_) => {}
+                Err(_) => return Err(vm.new_value_error("Error reading from Buffer".to_string())),
+            }
+        },
+        Some(read_byte) => {
+            bytes = vec![0; objint::get_value(&read_byte).to_usize().unwrap()];
+            match handle.read_exact(&mut bytes) {
```
Comment (coolreader18) [inline vm/src/stdlib/io.rs]: Same sorta thing,
```rust
handle.read_exact(&mut bytes).map_err(|| vm.new_value_error(...))?;
```

File: tests/snippets/test_imghdr.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    ('python.ras', 'rast'),
+    #('python.sgi', 'rgb'),
+    ('python.tiff', 'tiff'),
+    ('python.xbm', 'xbm'),
+    ('python.webp', 'webp'),
+    ('python.exr', 'exr'),
+)
+
+for fname, expected in TEST_FILES:
+    res = imghdr.what('tests/snippets/imghdrdata/'+fname)
```
Comment (coolreader18) [inline tests/snippets/test_imghdr.py]: Maybe use `__file__` here to figure out where the `imghdrdata` directory is.

**Comment by corona10** [comment]
@coolreader18 

Updated! Please take a look!

**Comment by coolreader18** [comment]
Nice work!

**Pull request merged**